### DISCUSSION TITLE
Not using full potential of package

### DISCUSSION THREAD

**Discussion by allanmcarvalho** [dm4t2/vue-currency-input#360] [Ideas]
Hello guy!

This package is awesome! Good job, pretty wrote, good Intl abstraction  and a lot of other gems...

But I don't understand why you wrote the component only for "currencies" when you already have a global package that has all the number management, including 'currencies'.

In my project I had to use it in more generic case, so the currency package will be a bad use full of improvisations to work properly. So I am implementing the same logic of this package but using your other package.

So the question: Why not create a more generic Intl vue component for numbers?

**Comment by NielsJorck** [comment]
I am having the exact same issue.
Great package, but it is a rather odd limitation when the difference between currencies and regular numbers is so small.
Granted most of my needs are in currency, but I do have examples with car mileage which is a number I would like to show with proper decimal and thousand seperation. 
Something incredibly similar to this, but as you state, having the define it as a "currency" would be a bad implementation as it isn't actually a currency.
Granted the package name does state it's intended purpose, but I haven't found a good alternative package so far.

**Comment by dm4t2** [comment]
Hi, thanks for your feedback! My library [Intl Number Input](https://github.com/dm4t2/intl-number-input) is probably what you are looking for. It's a framework-agnostic and generalized implementation, which can be easily [integrated with Vue](https://github.com/dm4t2/intl-number-input/blob/master/docs/.vitepress/theme/components/NumberInput.vue).

I currently working on an "official" Vue integration for Intl Number Input. Once it's published I will probably deprecate Vue Currency Input.

**Reply by NielsJorck** [reply]
I ended with the lazy route and used this package, just hiding the currency.
Sounds brilliant with the new package👍 I couldn't find a repo for it, still hidden I assume?

Just in case you didn't know a little FYI :)
Personally I found this package via vee-validate here: https://vee-validate.logaretm.com/v4/examples/value-formatting/
I did make a post on the discussion forum about their example as vue-currency-input actually does expose the numberValue now.
When you make the new package, it might be worthwhile giving them a poke for exposure :)
